### ISSUE TITLE
[Bug]: Invalid quota value on user's profile page

### ISSUE THREAD

**Issue by pboguslawski** [nextcloud/server#42695]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When user has quota set to 10GB (`10737418240` in their `ldapQuotaAttribute`) quota value on user's profile page (/settings/user) is different (8,5G). No such problem on files page (/apps/files/) nor active users page (/settings/users).

### Steps to reproduce

1. Set quota in LDAP to 10737418240 (10GB) for nc user.
2. Sync users from LDAP to nc.
3. See quota value on pages above.

### Expected behavior

User should see 10GB on their profile page (/settings/user).

### Installation method

None

### Nextcloud Server version

28

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [X] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by joshtrichards** [comment]
Was this on a 32-bit platform?

8.5 is quite close to difference between the maximum integer size in PHP on a 32-bit platform:

`10737418240 - 2147483648 = 8589934592`

**Comment by pboguslawski** [comment]
Same problem in nc v31.0.2 (Debian 12 64-bit):

* files

![Image](https://github.com/user-attachments/assets/21f3f196-6507-4da0-9378-d1c312ddbd22)

* profile (same account) 

![Image](https://github.com/user-attachments/assets/55862c27-1da3-4450-8179-d6dbbf10d07b)

**Comment by pboguslawski** [comment]
After changing quota in LDAP from 10 to 100G, after a while files shows updated value

![Image](https://github.com/user-attachments/assets/4476cdcd-0fc6-4f02-812d-e51b6539c263)

but profile still shows wrong quota

![Image](https://github.com/user-attachments/assets/e4cc56c2-067b-4c57-81de-2d8786604f20)

**Comment by nextcloud-command** [comment]
This issue has been automatically marked as stale because it has not had recent activity and seems to be missing some essential information. It will be closed if no further activity occurs. Thank you for your contributions.

**Comment by come-nc** [comment]
Hello,

According to my testing, the difference here is that the profile page takes the disk space into account, while the files page does not.
Can you double check if the disk where the home storage for this user is stored has less than 10GB of free space?

**Comment by pboguslawski** [comment]
> Can you double check if the disk where the home storage for this user is stored has less than 10GB of free space?

Yes, with quota set to 10G in LDAP for user, when `df -h` shows Avail=8.3G for FS with user data, profile page shows `You are using 0 B of 8.3 GB (0%)` and files shows `0 B of 10 GB used `.

Nextcloud should not display disk free info (it's admin job to keep enough avail storage) and should display quota in both places IMHO.

**Comment by come-nc** [comment]
> > Can you double check if the disk where the home storage for this user is stored has less than 10GB of free space?
> 
> Yes, with quota set to 10G in LDAP for user, when `df -h` shows Avail=8.3G for FS with user data, profile page shows `You are using 0 B of 8.3 GB (0%)` and files shows `0 B of 10 GB used `.
> 
> Nextcloud should not display disk free info (it's admin job to keep enough avail storage) and should display quota in both places IMHO.

Hum, in my opinion it’s the other way around, it should display free space in both places. It’s useless to know you have 5GB of free quota if you cannot upload a 5GB file.

The issue is if no quota is set, in this case I agree that we should not display disk space information to users 🤔 Unless really low.

**Comment by sorbaugh** [comment]
> > Can you double check if the disk where the home storage for this user is stored has less than 10GB of free space?
> 
> Yes, with quota set to 10G in LDAP for user, when `df -h` shows Avail=8.3G for FS with user data, profile page shows `You are using 0 B of 8.3 GB (0%)` and files shows `0 B of 10 GB used `.
> 
> Nextcloud should not display disk free info (it's admin job to keep enough avail storage) and should display quota in both places IMHO.

@marcoambrosini @jancborchardt wdyt? Should we display quota instead of free space?

**Comment by pboguslawski** [comment]
> Hum, in my opinion it’s the other way around, it should display free space in both places.

Disk free is internal infrastructure info that should not be exposed to end users. User should see only their quota and admin should take care of the rest. Nextcloud should not try to be nagios.

Both places should display same info.

If you still want to see df, please use extra option to turn it on (one setting for both places).

> It’s useless to know you have 5GB of free quota if you cannot upload a 5GB file.

This situation should not happen in correctly administered system (our issue was spotted in dev :).

**Comment by pboguslawski** [comment]
> The issue is if no quota is set

Cosider forcing default quota if not set. Or not displaying total avail if no quota is set. Displaying df makes DOS easier.

**Comment by jancborchardt** [comment]
@sorbaugh yes we should show quota, that makes more sense for people.


> > The issue is if no quota is set
> 
> Cosider forcing default quota if not set. Or not displaying total avail if no quota is set. Displaying df makes DOS easier.

@pboguslawski we should not force a default quota, that just complicates things for many cases to take care of this display issue.

Not displaying total available if no quota is set also seems strange. Isn't the main issue here that the values are not the same in Files and personal settings?

**Comment by pboguslawski** [comment]
> Isn't the main issue here that the values are not the same in Files and personal settings?

This is the original issue scope. Now the scope extended to displaying df also.

If default quota in case of no quota config is bad idea, consider displaying df the only if there is no quota config or new option explicitly says so (default new option value should be to display quota if configured IMHO).

**Comment by come-nc** [comment]
The percentage shown is already based on free space, when that is lower than quota. And this cannot be changed easily as it’s part of the API.

I opened a PR to align Files navigation with user settings and correctly show disk space when it’s lower than quota and a quota is set. https://github.com/nextcloud/server/pull/53693

This has nothing to do with df or nagios, Nextcloud already has the free space information and needs it for a number of reasons. If the UI shows 3GB available and uploading 1GB fails it will be seen as a bug.
